src/api/style.js

```diff
@@ -311,13 +312,17 @@ export default class Style {
         if (!(styleSpec.color instanceof Expression)) {
             throw new CartoValidationError('style', 'nonValidExpression[color]');
         }
-        if (!(styleSpec.width instanceof Expression)) {
-            throw new CartoValidationError('style', 'nonValidExpression[width]');
-        }
         if (!(styleSpec.strokeColor instanceof Expression)) {
             throw new CartoValidationError('style', 'nonValidExpression[strokeColor]');
         }
-        if (!(styleSpec.strokeWidth instanceof Expression)) {
+        if (util.isNumber(styleSpec.width)) {
```

Comment:
There is a `implicitCast()` function to do exactly this, please, use it.

Reply:
We need control here of the type to launch the proper exceptions so I can use `implicitCast(` instead of `new float(`, but the rest should be the same.

Reply:
I know, but `implicitCast()` should be the one generating the `float()` and the one checking the `isNumber()`

Reply:
In order to do that this would be the final code:

```js
if (!(styleSpec.width instanceof Expression)) {
    styleSpec.width = implicitCast(styleSpec.width);
    if (!(styleSpec.width instanceof Expression)) {
        throw new CartoValidationError('style', 'nonValidExpression[width]');
    }
}
```

But this would allow to use strings also for `width` and `strokeWidth`, and we want to avoid that. We need explicit cast instead.

Reply:
After talking about it we have decided to keep type checking for other PR.

We have also found that the conflictive code in `implicitCast` can be removed:
```js
if (!(value instanceof Expression) && value.type !== 'paletteGenerator' && value.type !== 'float') {
    throw new Error('value cannot be casted');
}
```

So I go ahead with it.